src/impl_x86__base_implementation.inl

```diff
@@ -570,6 +573,10 @@ X86Microarchitecture GetX86Microarchitecture(const X86Info* info) {
         return X86_UNKNOWN;
     }
   }
+  if (IsVendorByX86Info(info, CPU_FEATURES_VENDOR_ZHAOXIN_CTH) || 
+      IsVendorByX86Info(info, CPU_FEATURES_VENDOR_ZHAOXIN_SH)) {
+        return ZHAOXIN;
+  }
```

Comment:
same, could you separate?

Reply:
I have added support for ZHAOXIN product' s microarchitecture ZhangJiang, WuDaoKou, LuJiaZui and YongFeng.